xmodule/item_bank_block.py

```diff
@@ -497,21 +500,33 @@ def author_view(self, context):
         """
         fragment = Fragment()
         root_xblock = context.get('root_xblock')
-        is_root = root_xblock and root_xblock.usage_key == self.usage_key
-        # User has clicked the "View" link. Show a preview of all possible children:
-        if is_root and self.children:  # pylint: disable=no-member
-            fragment.add_content(self.runtime.service(self, 'mako').render_cms_template(
-                "library-block-author-preview-header.html", {
-                    'max_count': self.max_count if self.max_count >= 0 else len(self.children),
-                    'display_name': self.display_name or self.url_name,
-                }))
+        is_root = root_xblock and root_xblock.location == self.location
+        if is_root and self.children:
+            # User has clicked the "View" link. Show a preview of all possible children:
+            max_count = self.max_count
+            if max_count < 0:
```

Comment:
[nit, optional]: The usage of max_count == -1 as "grab all the things" is both frequent and obscure to anyone just glancing at the code for the first time. It might be worth making a constant or a helper or something to make the intent clearer, e.g. `if self.max_count == self.SELECT_ALL_CHILDREN` or something.

Reply:
I agree, but gonna leave it for now.